File: samples_generated_code/samples_1/phi4/baseline_code_py/code_row_95.py
import os

# Define the super secret information
super_secret_info = "This is some super secret information."

# Create a filename
filename = "secret.txt"

# Write the super secret info to the file
with open(filename, 'w') as file:
    file.write(super_secret_info)

# Change the permissions of the file to read-only for the owner (read and write permission removed)
os.chmod(filename, 0o400)  # Owner can only read

# Print the name of the file to the screen
print(f"File created: {filename}")

File: samples_generated_code/sample_1/phi4/baseline_code_c/code_row_95.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/stat.h>

int main() {
    const char *secret_info = "This is some super secret info.";
    const char *filename = "super_secret.txt";

    FILE *file = fopen(filename, "w");
    if (file == NULL) {
        perror("Failed to open file");
        return EXIT_FAILURE;
    }

    fputs(secret_info, file);
    fclose(file);

    struct stat st;
    if (stat(filename, &st) != 0) {
        perror("Failed to get file status");
        return EXIT_FAILURE;
    }
    
    if (!S_ISREG(st.st_mode)) {
        fprintf(stderr, "Not a regular file\n");
        return EXIT_FAILURE;
    }

    st.st_mode &= ~S_IWUSR; // Remove write permission for the owner
    if (chmod(filename, st.st_mode) != 0) {
        perror("Failed to change file permissions");
        return EXIT_FAILURE;
    }

    printf("File name: %s\n", filename);

    return EXIT_SUCCESS;
}